mantidproject/mantid
# Fix long mode in pearl powder diff scripts

**Description of work.**

Fix bug present in 6.2 and 6.3, where setting long_mode on in the configuration yaml file didn't result in long mode settings being activated in create_vanadium and focus actions

I've added a new unit test that tests this particular feature. I've also extended the PEARL system test to better test long mode. There was a test that ran a long mode focus but it's main aim was to check that subsequent focus actions with long mode off were correctly reverting the long mode setting.

**To test:**

Run the script on the linked issue and check that the workspace called PRL109803-109806_tt70_long_mods1-9 contains non-zero data between 4.5 and 8 angstroms.

To run this you'll need to unzip the following zip file somewhere and then update <folder> in the paths in the script with the location where you unzipped the files:

\\olympic\Babylon5\Public\DannyHindson\PEARLTestFilesPR33385.zip

Fixes #33383.


---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
